### ISSUE TITLE
Update Function ant01.TXT

### ISSUE THREAD

**Issue by gamersalpha** [colobot/colobot#1565]
there are several issues with the given code, here's a corrected version:
```

`extern void object::Attack()
{
    object p;
    float dist, prox;
    point nav1, nav2, dest;
    boolean advance = true;

    errmode(0); // do not stop if there is an error
    while (ismovie() != 0)
        wait(1);

    while (true)
    {
        p = radar(WingedShooter, 0, 360, 0, 50);
        if (p == null)
        {
            nav1.x = position.x - 5;
            nav1.y = position.y;
            nav1.z = position.z;
            nav2.x = position.x + 5;
            nav2.y = position.y;
            nav2.z = position.z;

            while (true)
            {
                goto(nav1);
                p = radar(WingedShooter, 0, 360, 0, 50);
                if (p != null)
                    break;

                goto(nav2);
                p = radar(WingedShooter, 0, 360, 0, 50);
                if (p != null)
                    break;
            }
        }
        else
        {
            dist = distance(p.position, position);
            if (dist <= 40 && !advance)
            {
                fire(p);
                advance = true;
            }
            else
            {
                prox = dist - 5;
                if (prox > 40)
                    prox = 40;
                if (prox < 5)
                    prox = 5;
                dest.x = (position.x - p.position.x) * prox / dist + p.position.x;
                dest.y = (position.y - p.position.y) * prox / dist + p.position.y;
                dest.z = (position.z - p.position.z) * prox / dist + p.position.z;
                goto(dest);
                advance = false;
            }
        }
        wait(1);
    }
}
```
`

Here are the changes made to the original code:

Added wait(1) at the end of the while loop to avoid overloading the CPU.
Added nav1.z and nav2.z to set the correct z-coordinate for navigation points.
Changed fire(p.position) to fire(p) to fire at the target object instead of its position.
Changed p.position.z to p.position in the calculation of dest to use the object's position instead of just the z-coordinate.

**Comment by hexagonrecursion** [comment]
* I will look into this after https://github.com/colobot/colobot/pull/1683 is merged

**Comment by hexagonrecursion** [comment]
It appears that ant01.TXT is not used in any missions in vanilla - all references to it were removed in colobot-gold-0.1.3-alpha https://github.com/colobot/colobot-data/commit/17d36f64baa6dfa5dec2a47a669bc23f9a2f39b1. Why do you want to change dead code? It should probably be deleted instead.
```sh
$ git -C data/ log -S ant01.txt
commit 17d36f64baa6dfa5dec2a47a669bc23f9a2f39b1
Author: krzys-h <krzys_h@interia.pl>
Date:   Fri Dec 27 19:52:42 2013 +0100

    Fix for colobot/colobot#268 - Removed prototype missions

commit fd64748ce470f5a08ae1577fc54daa76f0a1538c (tag: colobot-data-2012-09-13)
Author: Didier Raboud <odyx@debian.org>
Date:   Thu Sep 13 23:35:43 2012 +0200

    Import colobot-data 2012-09-13
```

> Added wait(1) at the end of the while loop to avoid overloading the CPU

This appears to be unnecessary.
1. If `radar(WingedShooter, 0, 360, 0, 50)` on line 13 returns `null`, we execute a goto() to a point 5 meters away - this takes some time - the wait is not necessary
2. If `radar(WingedShooter, 0, 360, 0, 50)` on line 13 returns non-null, we execute `fire(p)` every other iteration - this also takes some time - the wait is not necessary

> Added nav1.z and nav2.z to set the correct z-coordinate for navigation points.

Why? I thought goto() ignores the z coordinate.

> Changed fire(p.position) to fire(p) to fire at the target object instead of its position.

Syntax error: No function with this name accepts this kind of parameter

> Changed p.position.z to p.position in the calculation of dest to use the object's position instead of just the z-coordinate.

Where? The line `dest.z = (position.z-p.position.z)*prox/dist + p.position.z;` appears to be the same both in the original and in your modified version (not counting whitespace changes)
